e2e/kube/suite_test.go

```diff
@@ -16,28 +21,134 @@ func TestE2E(t *testing.T) {
 	RunSpecs(t, "E2E Kube Suite")
 }
 
+const (
+	cfOperatorRelease    = "cf-operator"
+	installTimeOutInSecs = "600"
+	helmCmd              = "helm"
+	kubeCtlCmd           = "kubectl"
+)
+
 var (
 	cliPath      string
 	env          *environment.Environment
 	stopOperator environment.StopFunc
 )
 
 var _ = BeforeSuite(func() {
-	var err error
-	cliPath, err = gexec.Build("code.cloudfoundry.org/cf-operator/cmd/cf-operator")
+	err := SetUpEnvironment()
 	Expect(err).ToNot(HaveOccurred())
-
-	env = environment.NewEnvironment()
-	stopOperator, err = env.Setup()
-	Expect(err).NotTo(HaveOccurred())
 })
 
 var _ = AfterSuite(func() {
-	if stopOperator != nil {
-		time.Sleep(3 * time.Second)
-		defer stopOperator()
+	err := TearDownEnvironment()
+	Expect(err).ToNot(HaveOccurred())
+})
+
+// TearDownEnvironment uninstall the cf-operator
+// related helm chart
+func TearDownEnvironment() error {
+	_, err := RunBinary(helmCmd, "delete", cfOperatorRelease, "--purge")
+	if err != nil {
+		return err
 	}
 
-	gexec.KillAndWait()
-	gexec.CleanupBuildArtifacts()
-})
+	err = DeleteOperatorResources()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// SetUpEnvironment ensures helm binary can run,
+// being able to reach tiller, and eventually it
+// will install the cf-operator chart.
+func SetUpEnvironment() error {
+
+	err := DeleteOperatorResources()
+	if err != nil {
+		return err
+	}
+
+	// Ensure tiller is there, if not
+	// then create it, via "init"
+	err = RunHelmBinaryWithCustomErr(helmCmd, "version", "-s")
+	if err != nil {
+		switch err := err.(type) {
+		case *CustomError:
+			if strings.Contains(err.StdOut, "could not find tiller") {
+				_, err := RunBinary(helmCmd, "init", "--wait")
+				if err != nil {
+					return err
+				}
+			}
+		default:
+			return err
+		}
+	}
+
+	dir, err := os.Getwd()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Install the helm chart in default ns
+	_, err = RunBinary(helmCmd, "install", fmt.Sprintf("%s%s", dir, "/../../helm/cf-operator"),
```

Comment:
I thought we need to run `make build-helm` first and install the tgz?

Reply:
it happens in the bin/test-kube-e2e